core/server/apps/amp/lib/helpers/amp_content.js

```diff
@@ -4,15 +4,43 @@
 // Turns content html into a safestring so that the user doesn't have to
 // escape it or tell handlebars to leave it alone with a triple-brace.
 //
-// Converts normal HTML into AMP HTML.
+// Converts normal HTML into AMP HTML with Amperize module and uses a cache to return it from
+// there if available. The cacheId is a combination of `updated_at` and the `slug`.
 
 var hbs             = require('express-hbs'),
-    ampContent;
+    Promise         = require('bluebird'),
+    Amperize        = require('amperize'),
+    moment          = require('moment'),
+    amperizeCache = {};
 
-ampContent = function () {
-    // var root = options.data.root;
-    // console.log('root:', root);
-    return new hbs.handlebars.SafeString(this.html);
-};
+function getAmperizeHTML(html, cacheId) {
+    if (!amperizeCache[cacheId]) {
+        return new Promise(function (resolve, reject) {
+            new Amperize().parse(html, function (err, res) {
```

Comment:
question: 
do we have to create a `new Amperize` instance for each requested amp template?
i think we can re-use the same for each request, can't we?